vault/resource_mongodbatlas_secret_role.go

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+func mongodbAtlasSecretRoleResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: mongodbAtlasSecretRoleCreateUpdate,
+		ReadContext:   ReadContextWrapper(mongodbAtlasSecretRoleRead),
+		UpdateContext: mongodbAtlasSecretRoleCreateUpdate,
+		DeleteContext: mongodbAtlasSecretRoleDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			consts.FieldBackend: {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Path where MongoDB Atlas secret backend is mounted",
+				ValidateFunc: provider.ValidateNoLeadingTrailingSlashes,
+			},
+			consts.FieldPath: {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Path where MongoDB Atlas backend is mounted",
+				ValidateFunc: provider.ValidateNoLeadingTrailingSlashes,
+			},
+			"name": {
```

Comment:
We have a constant for `name` in `consts.go`. If we don't have constants for any of the fields in this file, we should add them to `consts.go` and update the code to use constants here and in the tests. Thanks in advance for extending the constants! 🙏🏼

Reply:
Got it! Did find a field for `ttl`, the rest are added in `consts.go` and are being used in both files now.